### PR TITLE
feat: Adding raw blocks as leafs in `DirectoryBuilder`

### PR DESCRIPTION
**Pull request by ppodolsky**
In my case I need to manually craft directories, namely adding pre-existent nodes as children for new directory. Here are changes to support adding blocks in `DirectoryBuilder`

### CODE REVIEW

File: iroh-unixfs/src/builder.rs [hunk 1]
```diff
@@ -559,6 +598,10 @@ impl DirectoryBuilder {
 
         Ok(if let Some(path) = path {
             let mut dir = make_dir_from_path(path, chunker.clone(), degree).await?;
+            match &mut dir {
+                Directory::Basic(basic) => basic.entries.extend(entries),
+                Directory::Hamt(_) => unimplemented!(),
```
Comment (dignifiedquire) [inline iroh-unixfs/src/builder.rs:603]: `Hamt`s are implemented, so this shouldn't be the case

Comment (ppodolsky) [reply iroh-unixfs/src/builder.rs:603]: Ok, I need to figure out how to add entries to existent HamtNode. Probably worth to rebuild it entirely

Comment (rklaehn) [reply iroh-unixfs/src/builder.rs:603]: The way this builder works is a bit weird. If path is set, entries are ignored. So maybe you can just put in an assertion that either path is None or entries is empty and be done with it?

I tried to refactor all of this at some point, but was not entirely happy with the results.

Comment (ppodolsky) [reply iroh-unixfs/src/builder.rs:603]: Afaik there are no stoppers for working both, `path` and `entries`. It is just implemented this way for some reasons. I may try to refactor this part

Comment (ppodolsky) [reply iroh-unixfs/src/builder.rs]: @rklaehn Done with refactoring. I removed path variable in builder and added add_path method that walks over path, converts elements to `Entry` and adds them to `self.entries`. It makes API more consistent and allows to add multiple paths.

**Comment by ppodolsky** [comment]
I have refactored code here as you suggested @rklaehn. Take a look plz when you'll have a time

**Comment by ramfox** [comment]
Hello! The code that was previously hosted in this repository has been moved to [n0-computer/beetle](https://github.com/n0-computer/beetle). `beetle` is in maintenance mode, but will still accept PRs.

 If you are still interested in getting your PR merged, please re-open your PR on [n0-computer/beetle](https://github.com/n0-computer/beetle).

Check out our blog post for more info on our new direction for `iroh`: https://n0.computer/blog/a-new-direction-for-iroh/

**Pull request closed without merging**